### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by tnagler**
- New website is already online: www.vinecopulib.org.
- Docs can still be generated using  `make doc` (in standard doxygen layout).
- I shortened the README and refer to the relevant places for the full documentation.

### CODE REVIEW

File: docs/Doxyfile-mcss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@INCLUDE                = Doxyfile
+GENERATE_HTML           = NO
+GENERATE_XML            = Yes
+XML_PROGRAMLISTING      = NO
+M_FILE_TREE_EXPAND_LEVELS = 8
+M_SHOW_UNDOCUMENTED = NO
+M_LINKS_NAVBAR1 = setup \
```
Comment (tvatter) [inline docs/Doxyfile-mcss]: What does this do?

File: docs/Doxyfile.in [hunk 2]
```diff
@@ -764,7 +764,7 @@ WARN_LOGFILE           =
 # spaces.
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = @CMAKE_CURRENT_SOURCE_DIR@/src/ @CMAKE_CURRENT_SOURCE_DIR@/include/
+INPUT                  = @CMAKE_CURRENT_SOURCE_DIR@/src/ @CMAKE_CURRENT_SOURCE_DIR@/include/ @CMAKE_CURRENT_SOURCE_DIR@/docs/
```
Comment (tvatter) [inline docs/Doxyfile.in]: Why do we need to add this one?

Comment (tnagler) [reply docs/Doxyfile.in]: To make doxygen aware of the introductory articles (like [this](https://vinecopulib.github.io/vinecopulib/overview-bicop.html)), which are located in `docs/`.

File: docs/Doxyfile-mcss (hunk 1 above)
Comment (tnagler) [reply docs/Doxyfile-mcss]: That's the config file for the website.

**Pull request merged**